src/installer/installer.go

```diff
@@ -165,12 +166,37 @@ func (i *installer) InstallNode() error {
 	if err != nil {
 		i.log.Errorf("upload installation logs %s", err)
 	}
+	return i.finalize()
+}
+
+func (i *installer) finalize() error {
 	//update installation progress
 	i.UpdateHostInstallProgress(models.HostStageRebooting, "")
-	//reboot
-	if err = i.ops.Reboot(); err != nil {
+	if i.DryRunEnabled {
+		_, err := i.ops.ExecPrivilegeCommand(nil, "touch", i.Config.FakeRebootMarkerPath)
+		return errors.Wrap(err, "failed to touch fake reboot marker")
+	}
+
+	// in case ironic-agent exists on the host we should stop the assisted-agent service instead of rebooting the node.
+	// the assisted agent service stop will signal the ironic agent that we are done so that IPA can continue with its flow.
+	// regardless, we update the host install progress to `rebooting` since the node will get rebooted shortly after.
+	ironicAgentServiceName := "ironic-agent.service"
+	out, err := i.ops.ExecPrivilegeCommand(nil, "systemctl", "list-units", "--no-legend", ironicAgentServiceName)
```

Comment:
This check assumes that we will *always* have the socked mounted so that we can use `systemctl`. I think we already have it but just wanted to be explicit about it as I don't know if we already depend on it. I think.... we do

Reply:
The installer runs multiple other systemctl commands (it doesn't use a socket but have another trick instead :shushing_face: ), so no need to worry, this didn't add any dependency.